src/mpid/ch4/src/ch4_init.c

```diff
@@ -324,6 +324,15 @@ int MPID_Init(int *argc, char ***argv, int requested, int *provided, int *has_ar
         MPIR_ERR_SETANDJUMP1(mpi_errno, MPI_ERR_OTHER, "**pmi_kvs_get_my_name",
                              "**pmi_kvs_get_my_name %d", pmi_errno);
     }
+
+    if (has_parent) {
+        pmi_errno = PMI_KVS_Get(MPIDI_global.jobid, MPIDI_PARENT_PORT_KVSKEY,
+                                MPIDI_global.parent_port, MPIDI_MAX_KVS_VALUE_LEN);
+        if (pmi_errno != PMI_SUCCESS) {
+            MPIR_ERR_SETANDJUMP1(mpi_errno, MPI_ERR_OTHER, "**pmi_kvs_get_parent_port",
+                                 "**pmi_kvs_get_parent_port %d", pmi_errno);
+        }
+    }
```

Comment:
@yfguo Why do you split the spawn code apart?

Reply:
Just to put PMI stuff together.

Reply:
It's different functionality. `kvs_name` is needed everywhere, which should be fetched during init; while `parent_port` is only needed for dynamic processes. There are more PMI stuff scattered -- I am working on wrapping them. Don't worry about grouping them here. Just focus on functionality.